include/RAJA/policy/cuda/policy.hpp

```diff
@@ -757,6 +831,14 @@ struct get_index_thread<IndexGlobal<dim, BLOCK_SIZE, GRID_SIZE>>
 {
   using type = IndexGlobal<dim, BLOCK_SIZE, named_usage::ignored>;
 };
+///
+template <typename x_index, typename y_index, typename z_index>
+struct get_index_thread<IndexFlatten<x_index, y_index, z_index>>
+{
+  using type = IndexFlatten<typename get_index_thread<x_index>::type,
+                            typename get_index_thread<y_index>::type,
+                            typename get_index_thread<z_index>::type>;
+};
```

Comment:
We only need 3d?

Reply:
we can add 2d get_index_thread struct though if it would help in other applications.

Reply:
Is easy to add later if needed.